libs/cardano-protocol-tpraos/testlib/Test/Cardano/Protocol/Binary/Cddl.hs

```diff
@@ -0,0 +1,113 @@
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+{-# OPTIONS_GHC -Wno-redundant-constraints #-}
+
+module Test.Cardano.Protocol.Binary.Cddl (
+  cddlEraSpec,
+  huddleEraSpec,
+  postBabbageCddlSpec,
+  postBabbageHuddleSpec,
+) where
+
+import Cardano.Ledger.Babbage (BabbageEra)
+import Cardano.Ledger.Binary (Annotator, DecCBOR, EncCBOR)
+import Cardano.Ledger.Core
+import Cardano.Protocol.Crypto (StandardCrypto)
+import Cardano.Protocol.TPraos.OCert (OCert)
+import Test.Cardano.Ledger.Binary.Cddl (
+  CddlData,
+  cddlDecoderEquivalenceSpec,
+  cddlRoundTripAnnCborSpec,
+  cddlRoundTripCborSpec,
+ )
+import Test.Cardano.Ledger.Binary.Cuddle
+import Test.Cardano.Ledger.Common
+
+cddlEraSpec ::
+  forall era c bh bhbody.
+  ( Era era
+  , Eq (bh c)
+  , Show (bh c)
+  , DecCBOR (bh c)
+  , EncCBOR (bh c)
+  , DecCBOR (Annotator (bh c))
+  , Eq (bhbody c)
+  , Show (bhbody c)
+  , DecCBOR (bhbody c)
+  , EncCBOR (bhbody c)
+  ) =>
+  SpecWith CddlData
+cddlEraSpec = do
```

Comment:
I think adding a Block suffix will make it a bit more clear which spec we have in mind here
```suggestion
cddlEraBlockSpec = do
```